api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -124,6 +124,17 @@ message Bootstrap {
     LogFormat log_format = 1;
   }
 
+  message DeferredStatOptions {
+    // To save memory and CPU consumption on blocks of stats that are never referenced throughout
+    // the process lifetime, they can be encapsulated in a DeferredCreationCompatibleInterface. Then
```

Comment:
Stating the interface name seems to be very low-level.
Is the intent to have an attribute for a stat that declares whether it is deferred or not? If so, I suggest having a doc somewhere with a list of stats that are known to be deferred, and add a link here to the doc.

Reply:
Good point about the C++ class name -- that doesn't need to be in the API doc.

In this PR (per my request) there are no stats that are impacted by this setting; that will come in a follow-up PR with its own set of complexities. Adi you are saying that a doc of some form should be the place where we list what stats are affected? What kind of doc?

Reply:
Maybe an additional list isn't needed. Taking a step back, what is the expected behavior change from a user's perspective?

My assumption is that using a field and not a runtime flag implies that this is a knob that is intended to be kept.
Should a user care if some stat is part of the deferred list or not?

Reply:
According to an offline discussion with @stevenzzzz, there will be multiple PRs each adding a subset of stats that will be deferred.
It is unclear to me whether having a single flag to enable/disable all of the subsets is the right approach mainly because enabling the flag in two different versions of Envoy may present different behavior with the same traffic patterns.
Have you considered using different fields/list of enums for different subsets?

Assuming this was considered and was intentionally avoided, I suggest the following:
1) Setting this field as `[#not-implemented-hide:]` because setting it shouldn't change anything (correct me if I'm wrong).
2) clarifying the comment as follows:
```
When the flag is enabled, Envoy will lazily initialize a subset of the stats (see below). This will save memory and CPU cycles when creating the objects that own these stats, if those stats are never referenced throughout the lifetime of the process. However, it will incur additional memory overhead for these objects, and a small increase of CPU usage when a at least one of the stats is updated for the first time.

Groups of stats that will be lazily initialized:

* Cluster traffic stats: a subgroup of the [cluster statistics](add link) that are used when requests are routed to the cluster.
* ...
```